src/rtkcmn.c

```diff
@@ -1270,6 +1270,26 @@ extern int str2time(const char *s, int i, int n, gtime_t *t)
     *t=epoch2time(ep);
     return 0;
 }
+/* inversion of time2str (contrary to original str2time)------------------------
+* convert substring in string to gtime_t struct
+* args   : char   *s        I   string ("... yyyy/mm/dd hh:mm:ss ...")
+*          int    i,n       I   substring position and width
+*          gtime_t *t       O   gtime_t struct
+* return : status (0:ok,0>:error)
+*-----------------------------------------------------------------------------*/
+extern int str2time_(const char *s, int i, int n, gtime_t *t)
+{
+    double ep[6];
+    char str[256],*p=str;
+    
+    if (i<0||(int)strlen(s)<i||(int)sizeof(str)-1<i) return -1;
```

Comment:
Let's not stick to Tamoji's codestyle in our code! This is unreadable and unbearable :-(

Reply:
So in this particular line of code, I propose at least add spacing between operands and get rid of crazy casts to int as well.